storage/allocator_test.go

```diff
@@ -173,6 +173,27 @@ func createTestAllocator() (*stop.Stopper, *gossip.Gossip, *StorePool, allocator
 	return stopper, g, storePool, a
 }
 
+// mockStorePool sets up a collection of a alive and dead stores in the
+// store pool for testing purposes.
+func mockStorePool(storePool *StorePool, aliveStoreIDs, deadStoreIDs []proto.StoreID) {
+	storePool.mu.Lock()
+	defer storePool.mu.Unlock()
+
+	storePool.stores = make(map[proto.StoreID]*storeDetail)
+	for _, storeID := range aliveStoreIDs {
+		storePool.stores[storeID] = &storeDetail{
```

Comment:
it's not obvious to me why this map has pointer values; can you see about changing them to value types? if that's not possible, could you add a comment on storePool?